iOS_SDK/OneSignalSDK/Source/OneSignal.h

```diff
@@ -396,75 +410,69 @@ typedef NS_ENUM(NSUInteger, ONE_S_LOG_LEVEL) {
-+ (void)onesignal_Log:(ONE_S_LOG_LEVEL)logLevel message:(NSString*)message;
++ (void)onesignal_Log:(ONE_S_LOG_LEVEL)logLevel message:(NSString* _Nonnull)message;
 
 // - Tagging
-+ (void)sendTag:(NSString*)key value:(NSString*)value onSuccess:(OSResultSuccessBlock)successBlock onFailure:(OSFailureBlock)failureBlock;
++ (void)sendTag:(NSString* _Nonnull)key value:(NSString* _Nonnull)value onSuccess:(OSResultSuccessBlock _Nullable)successBlock onFailure:(OSFailureBlock _Nullable)failureBlock;
 + (void)sendTag:(NSString*)key value:(NSString*)value;
-+ (void)sendTags:(NSDictionary*)keyValuePair onSuccess:(OSResultSuccessBlock)successBlock onFailure:(OSFailureBlock)failureBlock;
-+ (void)sendTags:(NSDictionary*)keyValuePair;
-+ (void)sendTagsWithJsonString:(NSString*)jsonString;
-+ (void)getTags:(OSResultSuccessBlock)successBlock onFailure:(OSFailureBlock)failureBlock;
-+ (void)getTags:(OSResultSuccessBlock)successBlock;
-+ (void)deleteTag:(NSString*)key onSuccess:(OSResultSuccessBlock)successBlock onFailure:(OSFailureBlock)failureBlock;
-+ (void)deleteTag:(NSString*)key;
-+ (void)deleteTags:(NSArray*)keys onSuccess:(OSResultSuccessBlock)successBlock onFailure:(OSFailureBlock)failureBlock;
-+ (void)deleteTags:(NSArray*)keys;
-+ (void)deleteTagsWithJsonString:(NSString*)jsonString;
-// Optional method that sends us the user's email as an anonymized hash so that we can better target and personalize notifications sent to that user across their devices.
-// Sends as MD5 and SHA1 of the provided email
-+ (void)syncHashedEmail:(NSString*)email __deprecated_msg("Please refer to our new Email methods/functionality such as setEmail(). This method will be removed in a future version of the OneSignal SDK");
-
-// - Subscription and Permissions
-+ (void)IdsAvailable:(OSIdsAvailableBlock)idsAvailableBlock __deprecated_msg("Please use getPermissionSubscriptionState or addSubscriptionObserver and addPermissionObserver instead.");
++ (void)sendTags:(NSDictionary* _Nonnull)keyValuePair onSuccess:(OSResultSuccessBlock _Nullable)successBlock onFailure:(OSFailureBlock _Nullable)failureBlock;
++ (void)sendTags:(NSDictionary* _Nonnull)keyValuePair;
++ (void)sendTagsWithJsonString:(NSString* _Nonnull)jsonString;
++ (void)getTags:(OSResultSuccessBlock _Nullable)successBlock onFailure:(OSFailureBlock _Nullable)failureBlock;
++ (void)getTags:(OSResultSuccessBlock _Nullable)successBlock;
++ (void)deleteTag:(NSString* _Nonnull)key onSuccess:(OSResultSuccessBlock _Nullable)successBlock onFailure:(OSFailureBlock _Nullable)failureBlock;
++ (void)deleteTag:(NSString* _Nonnull)key;
++ (void)deleteTags:(NSArray* _Nonnull)keys onSuccess:(OSResultSuccessBlock _Nullable)successBlock onFailure:(OSFailureBlock _Nullable)failureBlock;
++ (void)deleteTags:(NSArray* _Nonnull)keys;
++ (void)deleteTagsWithJsonString:(NSString* _Nonnull)jsonString;
 
 + (OSPermissionSubscriptionState*)getPermissionSubscriptionState;
```

Comment:
` + (OSPermissionSubscriptionState*)getPermissionSubscriptionState;`
should we make the return type `_Nonnull`?

Reply:
added _nonnull